packages/dashboard/src/components/sidePanel/sections/baseSettingSection/index.tsx

```diff
@@ -35,39 +39,52 @@ export const BaseSettings: FC<Widget> = (widget) => {
       })
     );
   };
+
   const onXChange: (event: { detail: BaseChangeDetail }) => void = ({ detail: { value } }) =>
-    moveWidget({ x: parseNumber(value) - x, y: 0 });
+    moveWidget({ x: nonNaNParseInt(value) - x, y: 0 });
+
   const onYChange: (event: { detail: BaseChangeDetail }) => void = ({ detail: { value } }) =>
-    moveWidget({ y: parseNumber(value) - y, x: 0 });
-  const onWidthChange: (event: { detail: BaseChangeDetail }) => void = ({ detail: { value } }) =>
-    dispatch(
-      onResizeWidgetsAction({
-        anchor: 'right',
-        widgets: [widget],
-        vector: {
-          x: parseNumber(value) - width,
-          y: 0,
-        },
-      })
-    );
-  const onHeightChange: (event: { detail: BaseChangeDetail }) => void = ({ detail: { value } }) =>
-    dispatch(
-      onResizeWidgetsAction({
-        anchor: 'bottom',
-        widgets: [widget],
-        vector: {
-          y: parseNumber(value) - height,
-          x: 0,
-        },
-      })
-    );
+    moveWidget({ y: nonNaNParseInt(value) - y, x: 0 });
+
+  const onWidthChange: (event: { detail: BaseChangeDetail }) => void = ({ detail: { value } }) => {
+    if (isValidWidgetDimension(Number(value))) {
+      dispatch(
+        onResizeWidgetsAction({
+          anchor: 'right',
+          widgets: [widget],
+          vector: {
+            x: nonNaNParseInt(value) - width,
+            y: 0,
+          },
+        })
+      );
+    }
+
+    setValue('width', Number(value));
+  };
+
+  const onHeightChange: (event: { detail: BaseChangeDetail }) => void = ({ detail: { value } }) => {
+    if (isValidWidgetDimension(Number(value))) {
```

Comment:
Blocker: Use [`fieldState.invalid`](https://react-hook-form.com/api/usecontroller/controller/) to validate with controller rules, instead of validating with custom function.